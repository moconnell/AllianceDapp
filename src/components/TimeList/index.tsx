import TimeItem from "./components/TimeItem";
import {
  Container,
  InlineContainer,
  StyledSubtitle,
  StyledTitle,
  TimeContainer,
} from "./styles";
import { ReactComponent as Clock } from "../../assets/clock.svg";
import Time from "../../types/time";

interface TimeListProps {
  times: Time[];
  selectedTime?: Time;
  duration?: number;
  onChange?: (time: Time) => void;
}

const TimeList = ({
  times,
  onChange,
  selectedTime,
  duration,
}: TimeListProps) => (
  <Container>
    <StyledTitle>Select a time </StyledTitle>

    <InlineContainer>
      <InlineContainer>
        <Clock />
      </InlineContainer>

      {duration && <StyledSubtitle>{duration} min</StyledSubtitle>}
    </InlineContainer>

    <TimeContainer>
      {times.map((value: Time, index: any) => (
        <TimeItem
          active={
            value === selectedTime
          }
          onClick={(e) => {
            e.preventDefault();
            if (onChange) onChange(value);
          }}
          key={index + "key"}
          value={value}
        />
      ))}
    </TimeContainer>
  </Container>
);

export default TimeList;
