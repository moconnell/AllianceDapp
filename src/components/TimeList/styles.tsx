import styled from "styled-components";

export const TimeContainer = styled.div`
  width: 200px;
  max-width: 250px;
  display: flex;
  flex-direction: column;
  height: auto;
  overflow-y: scroll;
  scrollbar-color: #ff3864 transparent;
  padding-right: 10px;

  // For Google Chrome
  &::-webkit-scrollbar {
    border: 1px solid #ff3864;
    border-radius: 26px;
    width: 5px;
  }

  &::-webkit-scrollbar-thumb {
    background: #ff3864;
    border-radius: 26px;
    padding: 0.5px;
  }

  &::-webkit-scrollbar-track {
    background: transparent;
  }

  // For Internet Explorer
  & {
    scrollbar-face-color: #ff3864;
    scrollbar-track-color: transparent;
  }

  &::-moz-scrollbar {
    -moz-border: 1px solid #ff3864;
    border: 1px solid #ff3864;
    border-radius: 26px;
    width: 5px;
  }
  /* Track */

  &::-moz-scrollbar-track {
    background: transparent;
    border: 1px solid #ff3864;
    border-radius: 26px;
  }
  /* Handle */

  &::-moz-scrollbar-thumb {
    background: #ff3864;
    padding: 0.5px;
  }
`;

export const StyledSubtitle = styled.span`
  font-family: Inter;
  font-style: normal;
  font-weight: bold;
  font-size: 12px;
  line-height: 15px;
  display: flex;
  align-items: center;

  /* Raid Red */
  color: #ff3864;

  margin-bottom: 12px;
`;

export const Container = styled.div`
  padding: 0px 10px;
  display: flex;
  flex-direction: column;
  height: 350px;
  max-height: 350px;
`;

export const StyledTitle = styled.span`
  font-family: Inter;
  font-style: normal;
  font-weight: bold;
  font-size: 12px;
  line-height: 15px;
  display: flex;
  align-items: center;
  /* Raid Red */
  color: #ff3864;
  margin-bottom: 12px;
`;

export const InlineContainer = styled.div`
  margin-right: 4px;
  display: flex;
  margin-bottom: 12px;
`;
