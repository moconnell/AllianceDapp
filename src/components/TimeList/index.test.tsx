import { fireEvent, render } from "@testing-library/react";
import TimeList from "./index";
import { MOCK_TIMES } from "../../mock";

describe("<TimeList />", () => {
  const testData = [
    {},
    { selectedTime: MOCK_TIMES[0] },
    { selectedTime: MOCK_TIMES[0], duration: 30 },
  ];

  testData.forEach(({ selectedTime, duration }) =>
    it("should render correctly", async () => {
      const { asFragment } = render(
        <TimeList
          times={MOCK_TIMES}
          selectedTime={selectedTime}
          duration={duration}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    })
  );

  it("should pass value on change", async () => {
    const mockHandler = jest.fn();

    const { getByText } = render(
      <TimeList times={MOCK_TIMES} onChange={mockHandler} />
    );

    fireEvent.click(getByText("10:30 AM"));

    expect(mockHandler).toBeCalledWith({ hours: 10, minutes: 30 });
  });

  it("should ignore click when no onChange handler", async () => {
    const { getByText } = render(<TimeList times={MOCK_TIMES} />);

    fireEvent.click(getByText("10:30 AM"));
  });
});
