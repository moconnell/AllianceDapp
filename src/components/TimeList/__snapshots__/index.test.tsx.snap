// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeList /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-hKMtZM itofvw"
  >
    <span
      class="sc-eCYdqJ jXlBzm"
    >
      Select a time 
    </span>
    <div
      class="sc-jSMfEi gWnnJL"
    >
      <div
        class="sc-jSMfEi gWnnJL"
      >
        <svg>
          clock.svg
        </svg>
      </div>
    </div>
    <div
      class="sc-gsnTZi ecdcmO"
    >
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        10:00
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        10:30
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        11:00
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        11:30
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        12:00
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        13:00
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        13:30
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        14:00
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        14:30
      </button>
      <button
        class="sc-bczRLJ gfvuLZ"
        style="background-color: transparent;"
      >
        15:00
      </button>
    </div>
  </div>
</DocumentFragment>
`;
